drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,607 @@
+}
+
+/** Interrupt driven receiver enabling function */
+static void uarte_nrfx_irq_rx_enable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+}
+
+/** Interrupt driven receiver disabling function */
+static void uarte_nrfx_irq_rx_disable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+}
+
+/** Interrupt driven error enabling function */
+static void uarte_nrfx_irq_err_enable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ERROR_MASK);
+}
+
+/** Interrupt driven error disabling function */
+static void uarte_nrfx_irq_err_disable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ERROR_MASK);
+}
+
+/** Interrupt driven pending status function */
+static int uarte_nrfx_irq_is_pending(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	return ((nrf_uarte_int_enable_check(uarte,
+					    NRF_UARTE_INT_ENDTX_MASK) &&
+		 uarte_nrfx_irq_tx_ready_complete(dev))
+		||
+		(nrf_uarte_int_enable_check(uarte,
+					    NRF_UARTE_INT_ENDRX_MASK) &&
+		 uarte_nrfx_irq_rx_ready(dev)));
+}
+
+/** Interrupt driven interrupt update function */
+static int uarte_nrx_irq_update(struct device *dev)
+{
+	return 1;
+}
+
+/** Set the callback function */
+static void uarte_nrfx_irq_callback_set(struct device *dev,
+					uart_irq_callback_t cb)
+{
+	struct uarte_nrfx_data *data = get_dev_data(dev);
+
+	data->cb = cb;
+}
+
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+
+static const struct uart_driver_api uart_nrfx_uarte_driver_api = {
+	.poll_in		= uarte_nrfx_poll_in,
+	.poll_out		= uarte_nrfx_poll_out,
+	.err_check		= uarte_nrfx_err_check,
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill		= uarte_nrfx_fifo_fill,
+	.fifo_read		= uarte_nrfx_fifo_read,
+	.irq_tx_enable		= uarte_nrfx_irq_tx_enable,
+	.irq_tx_disable		= uarte_nrfx_irq_tx_disable,
+	.irq_tx_ready		= uarte_nrfx_irq_tx_ready_complete,
+	.irq_rx_enable		= uarte_nrfx_irq_rx_enable,
+	.irq_rx_disable		= uarte_nrfx_irq_rx_disable,
+	.irq_tx_complete	= uarte_nrfx_irq_tx_ready_complete,
+	.irq_rx_ready		= uarte_nrfx_irq_rx_ready,
+	.irq_err_enable		= uarte_nrfx_irq_err_enable,
+	.irq_err_disable	= uarte_nrfx_irq_err_disable,
+	.irq_is_pending		= uarte_nrfx_irq_is_pending,
+	.irq_update		= uarte_nrx_irq_update,
+	.irq_callback_set	= uarte_nrfx_irq_callback_set,
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+static int uarte_instance_init(struct device *dev,
+			       const struct uarte_init_config *config,
+			       u8_t interrupts_active)
+{
+	int err;
+	struct device *gpio_dev;
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+	struct uarte_nrfx_data *data = get_dev_data(dev);
+
+	gpio_dev = device_get_binding(CONFIG_GPIO_NRF5_P0_DEV_NAME);
```

Comment:
I overlooked it before. What if one uses a pin from P1?

Reply:
fixed, but not pretty